terraform/layer2-k8s/templates/cluster-autoscaler-values.yaml

```diff
@@ -10,10 +10,12 @@ autoDiscovery:
   clusterName: ${cluster_name}
 
 serviceMonitor:
-  enabled: false
+  enabled: true
+  interval: 10s
   namespace: monitoring
   selector:
-    prometheus: prometheus-operator
+    app: kube-prometheus-stack-operator
```

Comment:
It'd better configure prometheus operator to discover all servicemonitors instead of add labels for each servicemonitor. 

Add this in prometheus-values.yaml:
prometheus.prometheusSpec.serviceMonitorSelectorNilUsesHelmValues: false

Reply:
Thank you, I used `serviceMonitorSelectorNilUsesHelmValues: false` and removed all labels for service monitors.